app/src/userpages/components/ShareSidebar/Pagination.jsx

```diff
@@ -0,0 +1,124 @@
+import React, { useCallback, useState, useEffect } from 'react'
+import { Button } from '@streamr/streamr-layout'
+import styled, { css } from 'styled-components'
+import Sidebar from '$shared/components/Sidebar'
+import { MEDIUM } from '$shared/utils/styled'
+
+const PER_PAGE = 10
+
+const Buttons = styled.div`
+    transform: translateX(12px);
+
+    button {
+        width: 32px;
+    }
+`
+
+const Inner = styled.div`
+    display: flex;
+
+    > span {
+        flex-grow: 1;
+    }
+`
+
+const UnstyledArrow = ({ flip, ...props }) => (
+    <svg {...props} width="14" height="8" viewBox="0 0 14 8" xmlns="http://www.w3.org/2000/svg">
+        <path
+            // eslint-disable-next-line max-len
+            d="M0.811223 7.52969C0.51833 7.2368 0.518329 6.76193 0.811223 6.46903L6.47123 0.809025C6.76412 0.516132 7.239 0.516132 7.53189 0.809025L13.1887 6.46588C13.4816 6.75877 13.4816 7.23365 13.1887 7.52654C12.8959 7.81943 12.421 7.81943 12.1281 7.52654L7.00156 2.40002L1.87188 7.52969C1.57899 7.82259 1.10412 7.82259 0.811223 7.52969Z"
+            fill="currentColor"
+        />
+    </svg>
+)
+
+const Arrow = styled(UnstyledArrow)`
+    display: block;
+    transform: rotate(-90deg);
+
+    ${({ flip }) => !!flip && css`
+        transform: rotate(90deg);
+    `}
+`
+
+const subcollection = (collection, startAt) => (
+    collection.slice(startAt, startAt + PER_PAGE)
+)
+
+const UnstyledPagination = ({
+    collection = [],
+    onPage,
+    selectedUserId: selectedId,
+    onUserSelect,
+    ...props
+}) => {
+    const [page, setPage] = useState(0)
```

Comment:
Maybe could wrap `setPage` with a callback that clamps the bounds so that the callers don't have to worry about bounds checks?

```js
const setPage = useCallback((fn) => {
    unselect() // or maybe in an effect?
    setPageState((...args) => {
        const v = typeof fn !== 'function' ? fn(...args) : fn
        return Math.min(maxPage, Math.max(0, v))
    })
}, [setPageState, unselect])
```

then:

```js
const next = useCallback(() => setPage((current) => current + 1), [setPage])
const prev = useCallback(() => setPage((current) => current - 1), [setPage])
```

Maybe this is better? Debatable.